lib/logic.rb

```diff
@@ -0,0 +1,149 @@
+class Board
+  attr_accessor :board
+
+  def initialize
+    @turn = true
+    @board = [
+      [' ', ' ', ' '],
+      [' ', ' ', ' '],
+      [' ', ' ', ' ']
+    ]
+  end
+
+  def print_board
+    @board.each do |row|
+      puts row.to_s
+    end
+  end
+
+  def full?
+    @board.each do |row|
+      row.each do |value|
+        return false if value == ' '
+      end
+    end
+    true
+  end
+
+  def taken?(row, column)
+    if @board[row][column] != ' '
+      return true
+    end
+    false
+  end
+end
+
+class Player
+  attr_accessor :name, :symbol, :board
+  def initialize(name, symbol = nil)
+    @name = name
+    @symbol = symbol
+  end
+
+  def play_at(board)
```

Comment:
I observe that this method is mostly UI related. Therefore, I don't think it should reside inside the Player class. Let's separate any UI/display related logic from our classes